server/channels/app/platform/web_hub.go

```diff
@@ -196,11 +196,15 @@ func (ps *PlatformService) InvalidateCacheForChannelPosts(channelID string) {
 }
 
 func (ps *PlatformService) InvalidateCacheForUser(userID string) {
+	ps.InvalidateChannelCacheForUser(userID)
+	ps.Store.User().InvalidateProfileCacheForUser(userID)
+}
+
+func (ps *PlatformService) InvalidateChannelCacheForUser(userID string) {
 	ps.Store.Channel().InvalidateAllChannelMembersForUser(userID)
 	ps.ClearUserSessionCache(userID)
```

Comment:
Could you expand on why `ps.ClearUserSessionCache(userID)` was moved?

Reply:
It hasn't moved. What I have done is kept `InvalidateCacheForUser` to have the same functionality, but carved out `InvalidateChannelCacheForUser` from inside it.

In any case, `ClearUserSessionCache` wipes out the webConn channel member cache which is part of `InvalidateChannelCacheForUser`. So it's still correct.

Reply:
Thanks. I still don't understand what the session (`*model.Session`) has to do with channels :thinking: 

I find it rather unclear by reading these method calls why clearing the session would be needed on a channel invalidation instead of on the user invalidation. Maybe this is something we could improve since I suspect it would be extremely easy to mess it up without a whole knowledge of our internal caching logic.

Reply:
Again, the issue is that both are tied together.

```go
func (ps *PlatformService) ClearSessionCacheForUserSkipClusterSend(userID string) {
	ps.ClearUserSessionCacheLocal(userID)
	ps.invalidateWebConnSessionCacheForUser(userID)
}
```

`ClearUserSessionCacheLocal` doesn't have anything to do with channels. But `invalidateWebConnSessionCacheForUser` does. To split them out would be slightly out of scope, but should be done at some point.

Reply:
Okay, so by the sound of it, we are still invalidating more than we should. Would it be risky to extract the `ps.ClearUserSessionCacheLocal(userID)` call from there and move it in `ps.InvalidateCacheForUser` where it makes more sense?

Reply:
Yes, it would be risky. There was a nasty bug on exactly that area which we fixed: https://github.com/mattermost/mattermost/pull/27204#pullrequestreview-2089868305. These caches are very sensitive and there are hard to debug edge cases.